src/utilities/fastify/plugins/fastifyActiveEndpoint.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { jest } from '@jest/globals';
+import fastify from 'fastify';
+
+import { mockSpy } from '../../../testUtils/jest.js';
+import { configureMockEnvVars } from '../../../testUtils/envVars.js';
+import type { InternetEndpoint } from '../../awala/InternetEndpoint.js';
+import { MONGODB_URI } from '../../../testUtils/db.js';
+import { InternetEndpointManager } from '../../awala/InternetEndpointManager.js';
+
+import fastifyMongoose from './fastifyMongoose.js';
+import fastifyActiveEndpoint from './fastifyActiveEndpoint.js';
+
+const mockGetActiveEndpoint = mockSpy(jest.fn<() => Promise<InternetEndpoint>>());
+const mockMakeInitialSessionKey = mockSpy(jest.fn<() => Promise<void>>());
+const mockEndpointManagerInit = mockSpy(jest.spyOn(InternetEndpointManager, 'init'));
+
+describe('fastifyActiveEndpoint', () => {
+  configureMockEnvVars({
+    MONGODB_URI,
+  });
+
+  let mockActiveEndpoint: InternetEndpoint;
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const endpointManager: InternetEndpointManager = {
+      getActiveEndpoint: mockGetActiveEndpoint,
+    } as Partial<InternetEndpointManager> as any;
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    mockActiveEndpoint = {
+      makeInitialSessionKeyIfMissing: mockMakeInitialSessionKey,
+    } as Partial<InternetEndpoint> as any;
+
+    mockEndpointManagerInit.mockResolvedValueOnce(endpointManager);
+    mockGetActiveEndpoint.mockResolvedValueOnce(mockActiveEndpoint);
+  });
+
+  test('Fastify should be decorated with get active endpoint function', async () => {
+    const app = fastify();
+    await app.register(fastifyMongoose);
+
+    await app.register(fastifyActiveEndpoint);
+    await app.ready();
+
+    expect(app).toHaveProperty('getActiveEndpoint');
+    const activeEndpoint = await app.getActiveEndpoint();
+    expect(activeEndpoint).toBe(mockActiveEndpoint);
+  });
+
+  test('Active endpoint should be created once and cached', async () => {
```

Comment:
The title of the test was: 
```
Active endpoint should be created once and cached
```
You suggested to remove that. Maybe we still need a similar test?

NOTE: we are checking that the getActiveEnpoint was called once
`    expect(mockGetActiveEndpoint).toHaveBeenCalledOnce();`

Reply:
Remember that we're returning a function only as a workaround until #16 is implemented -- this way we can make sure that anything using the active endpoint can use it safely because it's all set up.

We won't need to return a function once #16 is implemented. It'd be a regular attribute, as it was originally: `fastify.activeEndpoint`. So, as an attribute, we wouldn't have to worry about `getActiveEndpoint()` being called multiple times.

TL;DR: This test is correct with the current version of the code, but I'll have to delete pretty soon as part of #16, so we might as well not push it.